test/e2e/apps/daemon_set.go

```diff
@@ -247,6 +247,13 @@ var _ = SIGDescribe("Daemon set [Serial]", func() {
 		Expect(err).NotTo(HaveOccurred(), "error failing a daemon pod")
 		err = wait.PollImmediate(dsRetryPeriod, dsRetryTimeout, checkRunningOnAllNodes(f, ds))
 		Expect(err).NotTo(HaveOccurred(), "error waiting for daemon pod to revive")
+
+		By("Check that the failed pod does not show up among existing daemon pods by inspecting the pods' UID.")
+		uid := pod.UID
+		podList = listDaemonPods(c, ns, label)
+		for _, pod = range podList.Items {
+			Expect(pod.UID).NotTo(Equal(uid))
```

Comment:
Maybe also check that the pod phase is "running" for all those found in the list?

Reply:
This has already been checked via `checkRunningOnAllNodes` call above, which links to the check whether the pod is available or not here: https://github.com/kubernetes/kubernetes/blob/master/test/e2e/apps/daemon_set.go#L550